fdmi/plugins/fdmi_app

```diff
@@ -60,19 +62,119 @@ MAX_KEYSTRS = 10240
 # is communicating.
 processes = []
 
+FDMI_producer = None
+ack_thread = None
+lock = threading.Lock()
+
+
+def send_ack_to_plugin(banner, p, rec_id):
+    print(banner + " ACK " + rec_id, flush = True)
+    ri = bytes('{}\n'.format(rec_id), encoding = "utf-8")
+    lock.acquire()
+    p.stdin.write(ri)
```

Comment:
I thought that sending ack to plugin will be inside the FDMI app, calling a func to ack events. I am not sure how writing to stdin is involved here. Can you elaborate a little bit about the workflow?

Reply:
plugin is written in C, and compiled to a Linux binary.
It is called by this python script.
This python script spawns the plugin.
The plugin sends FDMI records to stdout, and this python script reads from a pipe, which connects to the plugin's stdout.
This python script sends the FDMI events to kafka. When the python script is sure that the FDMI records are persistent, it sends the record ID back to plugin. It write to a pipe, which connects to the plugin's stdin.
When the plugin gets the record ID, it calls some API to release it.

plugin source (Motr service)  <-----RPC-----> user plugin <----pipe/{stdin,stdout,stderrr}------> fdmi_app  <----python kafka -------> kafka server.

Reply:
I see. Maybe we should figure out a better way to connect these two application and get rid of the pipe in the future.

Reply:
Let's discuss this offline.